src/System.Security.Principal.Windows/src/System/Security/Principal/WindowsIdentity.cs

```diff
@@ -208,20 +217,72 @@ public WindowsIdentity(string sUserPrincipalName)
 
                             if (profileBuffer != null)
                                 profileBuffer.Dispose();
-
-                            _safeTokenHandle = accessTokenHandle;
                         }
                     }
                 }
             }
         }
 
-        private static SafeLsaHandle ConnectToLsa()
+        /// <summary>
+        /// To obtain SafeLsaHandle we try before for an Impersonation token that require SeTcbPrivilege, if fails we fallback
+        /// to an Identification token
+        /// </summary>
+        /// <param name="logonProcessNameBuffer"></param>
+        /// <returns></returns>
+        private static SafeLsaHandle GetLsaHandle(byte[] logonProcessNameBuffer)
         {
             SafeLsaHandle lsaHandle;
-            int ntStatus = Interop.SspiCli.LsaConnectUntrusted(out lsaHandle);
-            if (ntStatus < 0) // non-negative numbers indicate success
-                throw GetExceptionFromNtStatus(ntStatus);
+
+            try
+            {
+                // Try to get an impersonation token
+                Privilege privilege = null;
+                try
+                { 
+                    // Try to enable the TCB privilege if possible
+                    try
+                    {
+                        privilege = new Privilege("SeTcbPrivilege");
+                        privilege.Enable();
+                    }
+                    catch (PrivilegeNotHeldException) { }
+
+                    ushort sourceNameLength = checked((ushort)(logonProcessNameBuffer.Length));
+                    using (SafeLocalAllocHandle sourceNameBuffer = Interop.Kernel32.LocalAlloc(0, new UIntPtr(sourceNameLength)))
+                    {
+                        if (sourceNameBuffer.IsInvalid)
+                            throw new OutOfMemoryException();
```

Comment:
Missing braces (this if is contained within a braced block, therefore braces are required).